teacher_student_dynamics/data_modules/hidden_manifold.py

```diff
@@ -37,7 +39,17 @@ def __init__(
 
         self._activation_name = activation
         self._activation = self._get_activation_function()
-        self._feature_matrix = feature_matrix # DxN matrix
+
+        if rotation_matrix is not None:
+            self._unrotated_feature_matrix = feature_matrix.to(torch.float32)
+            if zero_matrix is not None:
+                feature_matrix = torch.vstack((feature_matrix, zero_matrix)).to(
+                    torch.float32
+                )
+            self._feature_matrix = feature_matrix.T.mm(rotation_matrix)
```

Comment:
Similar here. Wouldn't `unrotated_feature_matrix` be multiplied by the rotation matrix?